### DISCUSSION TITLE
How can test code with CONNECT_COUNT > ${client_number} work?

### DISCUSSION THREAD

**Discussion by GrasshopperBears** [ANLAB-KAIST/KENSv3#54] [Project Q&A 2. Transport Layer (TCP)] [answered]
When we see the test case, let's say TestEnv_Reliable.TestAccept_Backlog1, there are 6 clients and ACCEPT_COUNT is 10.
As far as I know, `accept()` should be blocked if there's no pending connection, so after dealing with 3 clients(backlog=3), returning 3 times of accept call, I think 4th `accept()` call cannot be returned.
Then how the test code can work with this case?

**Comment by Pusnow** [comment, marked as answer]
In the test case, there is `EXPECT_ACCEPT` value. 

https://github.com/ANLAB-KAIST/KENSv3/blob/0226d32cc49a1a2df8c2a3d613ba8e7509502549/app/kens/testhandshake.cpp#L205-L207

> Then how the test code can work with this case?
Test code will counts successful `accept` and compare it with the  `EXPECT_ACCEPT` value.

**Reply by GrasshopperBears** [reply]
Thank you 👍
